src/main/java/org/amahi/anywhere/activity/AuthenticationActivity.java

```diff
@@ -203,6 +231,11 @@ public void afterTextChanged(Editable editable) {
 
 		}
 		else {
+            imageView.setVisibility(imageView.VISIBLE);
+
+            InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
+            imm.hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), InputMethodManager.RESULT_UNCHANGED_SHOWN);
+
```

Comment:
If you are using these same three lines both in the `if` block and the `else` block. Then why not just add them outside the if / else , only once!